sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTargetTest.java

```diff
@@ -436,4 +442,87 @@ public void eventPropertiesCascading() {
         expectedProperties.put("g", "7777");
         assertEquals(expectedProperties, log.getProperties());
     }
+
+    /* Property Configurator tests */
+    @Test
+    public void setCommonSchemaProperties() {
+        CommonSchemaLog log = new CommonSchemaEventLog();
+        log.setExt(new Extensions());
+        log.getExt().setApp(new AppExtension());
+
+        /* Get property configurator and set properties. */
+        PropertyConfigurator pc = Analytics.getTransmissionTarget("test").getPropertyConfigurator();
+        pc.setAppVersion("appVersion");
+        pc.setAppName("appName");
+        pc.setAppLocale("appLocale");
+        pc.onPreparingLog(log, "groupName");
+
+        /* Assert properties set on common schema. */
+        assertEquals("appVersion", log.getExt().getApp().getVer());
+        assertEquals("appName", log.getExt().getApp().getId());
+        assertEquals("appLocale", log.getExt().getApp().getLocale());
+    }
+
+    @Test
+    public void inheritCommonSchemaProperties() {
+        CommonSchemaLog log = new CommonSchemaEventLog();
+        log.setExt(new Extensions());
+        log.getExt().setApp(new AppExtension());
+
+        /* Set properties on parent to override unset properties on child */
+        AnalyticsTransmissionTarget parent = Analytics.getTransmissionTarget("parent");
+        parent.getPropertyConfigurator().setAppVersion("appVersion");
+        parent.getPropertyConfigurator().setAppName("appName");
+        parent.getPropertyConfigurator().setAppLocale("appLocale");
+
+        AnalyticsTransmissionTarget child = parent.getTransmissionTarget("child");
+        child.getPropertyConfigurator().onPreparingLog(log, "groupName");
+
+        /* Assert properties set on common schema. */
+        assertEquals("appVersion", log.getExt().getApp().getVer());
+        assertEquals("appName", log.getExt().getApp().getId());
+        assertEquals("appLocale", log.getExt().getApp().getLocale());
+    }
+
+    @Test
+    public void commonSchemaPropertiesNotSetWhenDisabled() {
+        CommonSchemaLog log = new CommonSchemaEventLog();
+        log.setExt(new Extensions());
+        log.getExt().setApp(new AppExtension());
+
+        /* Get target, disable it, and set properties. */
+        AnalyticsTransmissionTarget target = Analytics.getTransmissionTarget("test");
+        target.setEnabledAsync(false);
```

Comment:
add .get() to each setEnabledAsync calls to make sure its synced (to wait on the future task to complete)